strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ArtifactEntryServiceImpl.java

```diff
@@ -1,91 +1,83 @@
 package org.carlspring.strongbox.services.impl;
 
-import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
-import org.carlspring.strongbox.domain.ArtifactEntry;
-import org.carlspring.strongbox.repository.ArtifactRepository;
-import org.carlspring.strongbox.services.ArtifactEntryService;
-
-import javax.inject.Inject;
 import java.util.Collections;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
-import com.orientechnologies.orient.core.query.OQueryAbstract;
-import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
-import com.orientechnologies.orient.object.db.OObjectDatabaseTx;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
+import org.carlspring.strongbox.data.service.CommonCrudService;
+import org.carlspring.strongbox.domain.ArtifactEntry;
+import org.carlspring.strongbox.services.ArtifactEntryService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.cache.CacheManager;
 import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 
+import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
+
 /**
  * DAO implementation for {@link ArtifactEntry} entities.
  *
  * @author Alex Oreshkevich
  */
 @Service
 @Transactional
-class ArtifactEntryServiceImpl
-        implements ArtifactEntryService
+class ArtifactEntryServiceImpl extends CommonCrudService<ArtifactEntry> implements ArtifactEntryService
 {
 
     private static final Logger logger = LoggerFactory.getLogger(ArtifactEntryService.class);
 
     // will help us avoid to have hardcoded name of this class
     private static final String ARTIFACT_ENTRY_CLASS_NAME = ArtifactEntry.class.getSimpleName();
```

Comment:
We have ArtifactEntry.class call twice (here and in getEntityClass()), could we store it in some variable?